### DISCUSSION TITLE
[Solved] How to rewrite my settings.json to use ruff as codefixer and black formattng rules?

### DISCUSSION THREAD

**Discussion by simkimsia** [astral-sh/ruff#7474] [Q&A] [answered]
this is my current workspace settings.json

```
{
    "[python]": {
      "editor.defaultFormatter": "ms-python.black-formatter",
      "editor.formatOnSave": true,
      "editor.codeActionsOnSave": {
        "source.fixAll.ruff": true,
        "source.organizeImports.ruff": true
      },
    },
    "python.linting.flake8Enabled": true,
    "python.linting.enabled": true,
    "python.linting.pylintEnabled": false,
    "python.sortImports.args": [
        "--profile=black"
    ],
    "python.linting.flake8Args": [
      "--max-line-length=88",
      "--select=E,W,F",
      "--ignore=E203,W503"
    ],
}

```

right now, everything else outside of "[python]" is marked as to be deprecated

i like to enforce the 88 char max line length, the ignoring of E203, W503, and selecting of E, W, Fand skip removing unused imports for __init__.py and have the sorting of imports be done using black.

How do I rewrite the to be deprecated settings?

**Comment by JonathanPlasse** [comment, marked as answer]
```json
{
    "[python]": {
      "editor.defaultFormatter": "ms-python.black-formatter",
      "editor.formatOnSave": true,
      "editor.codeActionsOnSave": {
        "source.fixAll": "always",
      }
    },
    "ruff.args": ["--select=E,W,F,I001,PL", "--ignore=E203", "--per-file-ignores=**/__init__.py:F401"]
}
```
I invite you to read the [Ruff](https://docs.astral.sh/ruff/) documentation, and its [VSCode extension](https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff) for further details.

**Reply by simkimsia** [reply]
No offense, but neither the doc nor the extension readme contains information about `ruff.args` 

<img width="981" alt="image" src="https://github.com/astral-sh/ruff/assets/245021/49ed6fac-339c-4942-a5ce-4473b9696a63">
<img width="689" alt="image" src="https://github.com/astral-sh/ruff/assets/245021/ebfe6df9-d034-488c-a06e-6d57ac823849">

I do see a 
<img width="822" alt="image" src="https://github.com/astral-sh/ruff/assets/245021/8e0fef2c-080c-4048-a68d-0db5207780b3">


But as a newbie, it wouldn't have occurred to me to use it like 
```
"ruff.args": ["--select=E,W,F,I001,PL", "--ignore=E203", "--per-file-ignores=**/__init__.py:F401"]
```

exactly.

Thanks to your answer, I now learn this.

Thank you, I also now found https://docs.astral.sh/ruff/rules/ which lists out all the various rule selection figures for those who stumble up on this in future.

**Reply by JonathanPlasse** [reply]
Would you be interested in making a pull request to help future newbies?

**Reply by simkimsia** [reply]
Sure

How do I do that?

Sorry I mean which part of the documentation should I write the above in?

I know how to make a PR via GitHub

**Reply by JonathanPlasse** [reply]
Where did you expect to find it?
Personally, I would add your use of `ruff.args` as an example of configuration in the VSCode extension documentation and link to the [command line interface reference](https://docs.astral.sh/ruff/configuration/#command-line-interface) to find other argument options.

**Reply by simkimsia** [reply]
so if I make a PR on the README on https://github.com/astral-sh/ruff-vscode that would do it?

**Reply by simkimsia** [reply]
I made a PR https://github.com/astral-sh/ruff-vscode/pull/283 as per suggested. I'm glad I did it. Thanks for the encouragement, @JonathanPlasse

**Reply by JonathanPlasse** [reply]
You are welcome
